selendroid-standalone/src/main/java/io/selendroid/standalone/server/grid/SelfRegisteringRemote.java

```diff
@@ -47,6 +51,12 @@ public SelfRegisteringRemote(SelendroidConfiguration config, SelendroidStandalon
     this.driver = driver;
   }
 
+  public void performRegistrationIfNotRegistered(final boolean firstTime) throws Exception {
+    if (firstTime || !isAlreadyRegistered()){
```

Comment:
Maybe it would be better to make this class aware of whether this happens the first time or not (make `firstTime` a bool here).